newIDE/app/src/MainFrame/index.js

```diff
@@ -1507,6 +1508,15 @@ const MainFrame = (props: Props) => {
           i18n: i18n,
           project: state.currentProject,
           onChooseProject: chooseProject,
+          onOpenRecentFile: (fileMetadata: FileMetadata) => {
+            openFromFileMetadata(fileMetadata).then(state => {
```

Comment:
We're missing one last - but important - "detail". Turns out that openFromFileMetadata will open the fileMetadata passed as argument with the current "storage provider". 
On the Electron app, it's simple: there is a unique storage provider, which is `LocalFileStorageProvider`.
But on the web-app, which might have a MainMenu in the future, we can have mutliple storage providers. One FileMetadata can have been opened with "GoogleDrive", another one with "Dropbox", another one with "OneDrive". 
This means that we must:
1. Know the name of the storage where the file was originally saved.
2. Find this storage in our list of storage providers.
3. Tell GDevelop to use it before calling openFromFileMetadata (`openFromFileMetadata` will use the current one).

For 1, this means that instead of FileMetadata, you must pass something that is `FileMetadataAndStorageProviderName`. I recommend something like `type FileMetadataAndStorageProviderName = {| fileMetadata: FileMetadata, storageProviderName: string |}`.
Using this, it's easy to do 2: in the `props`  of `MainFrame`, get `storageProviders`, filter them to find the one with `internalName` === `storageProviderName`.
If none is found, use showErrorBox to tell the user that this file can't be opened as GDevelop does not know how to open this file. (should not happen - but better be defensive!)
Finally, `getStorageProviderOperations` pass it as argument the `storageProvider` you found - this will change the storage provider being used (I know, `getXXX` is a terrible name, have to change it). In the promise, call `openFromFileMetadata`.

This means that you have to update the whole context and most part of the PR to use FileMetadataAndStorageProviderName (sorry, a bit of refactoring 😅).. and this means that when you call `insertRecentProjectFile`, you have to pass the FileMetadata and the internal name of the storage provider. 
How to get this internal name? I recommend modifying `ProjectStorageProviders` to expose `getStorageProvider` (in addition to `getStorageProviderOperations`), which returns `this.state.currentStorageProvider || emptyStorageProvider`.

And that should be it! Note there is no risk of stale props because the props are getter functions that are not changing when we change the storage provider.

How to test all of this. Check that it's working on the electron app and I can check the web-app.

Give it a try but if it's too complicated I can do this myself - leave in this case a TODO here.

Reply:
@4ian your description of the way to call `openFromFileMetadata()` seems very similar to `chooseProjectWithStorageProviderPicker()` without the picker. So, more or less I have to store the name of `StorageProvider` and call a modified `chooseProjectWithStorageProviderPicker()` right?

Reply:
>  So, more or less I have to store the name of StorageProvider and call a modified chooseProjectWithStorageProviderPicker() right?

Yes that's it - maybe a new function called "openFromFileMetadataWithStorageProvider", something like this.